src/Microsoft.CodeQuality.Analyzers/Core/QualityGuidelines/MarkMembersAsStatic.cs

```diff
@@ -227,7 +229,7 @@ private static bool IsExplicitlyVisibleFromCom(IMethodSymbol methodSymbol, Compi
             return false;
         }
 
-        private static ImmutableArray<INamedTypeSymbol> GetSkippedAttributes(Compilation compilation)
+        private static ImmutableArray<INamedTypeSymbol> GetSkippedAttributes(WellKnownTypeProvider wellKnownTypeProvider)
         {
             ImmutableArray<INamedTypeSymbol>.Builder? builder = null;
```

Comment:
Nit: can simplify this method by using the pooled `ArrayBuilder` instead.

Reply:
I am not entirely sure about the usage, could you review the commit?

Reply:
I didn't managed to use the pooled `ArrayBuilder` correctly. Am I supposed to replace 

```csharp
builder ??= ImmutableArray.CreateBuilder<INamedTypeSymbol>();
builder.Add(symbol);
```

with

```csharp
ArrayBuilder<INamedTypeSymbol>.GetInstance().Add(symbol);
```

Because when doing so the builder doesn't contain all the types.

Please let me know the proper use of this builder and I'll update this PR.

Reply:
Its fine to keep the current code, we can do a separate PR to audit all uses of ImmutableArray and ImmutableDictionary builders.